tensorrt_llm/_torch/pyexecutor/executor_request_queue.py

```diff
@@ -563,13 +571,26 @@ def _broadcast_new_requests(
 
         # Tag for communication
         tag = self.dist.pp_size  # Use pp_size as tag to avoid conflicts
+        works = []
 
         # Send payloads
         if not self.dist.is_first_pp_rank:
             payloads = self.dist.recv_object(self.dist.prev_pp_rank, tag)
 
         if not self.dist.is_last_pp_rank:
-            self.dist.send_object(payloads, self.dist.next_pp_rank, tag)
+            if self._disable_mpi:
+                works.extend(
+                    self.dist.isend_object(payloads, self.dist.next_pp_rank,
+                                           tag))
+            else:
+                self.dist.send_object(payloads, self.dist.next_pp_rank, tag)
+
+        for work in works:
+            try:
+                work.wait()
+            except Exception as e:
+                raise RuntimeError(
+                    f"Asynchronous broadcast operation failed: {e}") from e
```

Comment:
Not sure I understand why we need the `works` list here, looks like only one item can be added. Could we refactor as:

```suggestion
            if self._disable_mpi:
                isend_payload = 
                    self.dist.isend_object(payloads, self.dist.next_pp_rank,
                                           tag)
                try:
                    isend_payload.wait()
                except Exception as e:
                    raise RuntimeError(
                        f"Asynchronous broadcast operation failed: {e}") from e                           
            else:
                self.dist.send_object(payloads, self.dist.next_pp_rank, tag)
```

Reply:
Updated. @pcastonguay PTAL, thanks.